database/schema.sql

```diff
@@ -207,6 +207,72 @@ CREATE TABLE rotation_move_history (
   FOREIGN KEY (new_bucket) REFERENCES rotation_buckets (bucket_id)
 );
 
+-- BUDGET TABLES
+
+-- Table of all accounts Ruddock owns.
+CREATE TABLE budget_accounts (
+  account_id INTEGER NOT NULL AUTO_INCREMENT,
+  account_name VARCHAR(32) NOT NULL,
+  initial_balance NUMERIC(9,2) NOT NULL, -- this is sufficent to store up to 10 million
+  PRIMARY KEY (account_id)
+);
+
+-- Table of fiscal years and when they start.
+CREATE TABLE budget_fyears (
+  fyear_id INTEGER NOT NULL AUTO_INCREMENT,
+  fyear_num INTEGER NOT NULL, -- the number itself, like 2016
+  start_date DATE NOT NULL,
+  end_date DATE NOT NULL,
+  PRIMARY KEY (fyear_id)
+);
+
+-- Table of all budgets.
+CREATE TABLE budget_budgets (
+  budget_id INTEGER NOT NULL AUTO_INCREMENT,
+  budget_name VARCHAR(32) NOT NULL,
+  fyear_id INTEGER NOT NULL,
+  starting_amount NUMERIC(9,2) NOT NULL,
```

Comment:
It just has the starting amount, since the current amount can be computed from the transactions. I didn't record it in the database because I thought the two could get out of sync. I guess I could make a view holding the current amount as well. Thoughts?